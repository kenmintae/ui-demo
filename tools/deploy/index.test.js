const { generateStorybookDeployConfig } = require('./');

describe('Given the correct process.env for a production deployTarget', () => {
  beforeAll(() => {
    delete process.env.AWS_ACCESS_KEY_ID;
    delete process.env.AWS_SECRET_ACCESS_KEY;
    delete process.env.GITHUB_TOKEN;
    delete process.env.BUILD_SHA;
    delete process.env.NOTIFICATION_SHA;
    process.env.AWS_ACCESS_KEY_ID = 'FAKE_AWS_ACCESS_KEY_ID';
    process.env.AWS_SECRET_ACCESS_KEY = 'FAKE_AWS_SECRET';
    process.env.GITHUB_TOKEN = 'FAKE_GITHUB_TOKEN';
    process.env.TRAVIS_PULL_REQUEST_SHA = 'FAKE_TRAVIS_PULL_REQUEST_SHA';
    process.env.TRAVIS_COMMIT = 'FAKE_TRAVIS_COMMIT';
  });
  describe('generateStorybookDeployConfig', () => {
    it('matches snapshot using travis env vars', () =>
      expect(generateStorybookDeployConfig('production')).toMatchSnapshot());
    it('matches snapshot using gondor env vars', () => {
      process.env.BUILD_SHA = 'FAKE_SHA';
      process.env.NOTIFICATION_SHA = 'FAKE_NOTIFICATION_SHA';
      expect(generateStorybookDeployConfig('production')).toMatchSnapshot();
    });

    it('uses the correct s3prefix', () => {
      const config = generateStorybookDeployConfig('production');
      expect(config.s3.prefix).toEqual('app/stories');
      expect(config['s3-index'].prefix).toEqual('app/stories');
    });
  });
});
