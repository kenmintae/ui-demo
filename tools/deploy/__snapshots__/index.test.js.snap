// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the correct process.env for a production deployTarget generateStorybookDeployConfig matches snapshot using gondor env vars 1`] = `
Object {
  "build": Object {
    "environment": "production",
    "outputPath": "./storybook-static/",
  },
  "display-revisions": Object {
    "amount": 10,
    "revisions": [Function],
  },
  "github-deployment-status": Object {
    "environment": "stories",
    "org": "zenefits",
    "ref": "FAKE_NOTIFICATION_SHA",
    "repo": "z-frontend",
    "targetUrl": "http://ui.zenefits.com/app/stories/index.html:FAKE_SHA",
    "token": "FAKE_GITHUB_TOKEN",
  },
  "pipeline": Object {
    "alias": Object {
      "s3-index": Object {
        "as": Array [
          "s3-index",
          "s3-iframe",
        ],
      },
    },
  },
  "revision-data": Object {
    "type": "git-commit",
  },
  "s3": Object {
    "accessKeyId": "FAKE_AWS_ACCESS_KEY_ID",
    "bucket": "ui.zenefits.com",
    "filePattern": "{**,docs}/*.{js,css,png,gif,ico,jpg,map,xml,txt,svg,swf,eot,ttf,woff,woff2,otf}",
    "prefix": "app/stories",
    "region": "us-east-1",
    "secretAccessKey": "FAKE_AWS_SECRET",
  },
  "s3-iframe": Object {
    "accessKeyId": "FAKE_AWS_ACCESS_KEY_ID",
    "allowOverwrite": true,
    "bucket": "ui.zenefits.com",
    "filePattern": "iframe.html",
    "prefix": "app/stories",
    "region": "us-east-1",
    "revisionKey": "FAKE_SHA",
    "secretAccessKey": "FAKE_AWS_SECRET",
  },
  "s3-index": Object {
    "accessKeyId": "FAKE_AWS_ACCESS_KEY_ID",
    "allowOverwrite": true,
    "bucket": "ui.zenefits.com",
    "prefix": "app/stories",
    "region": "us-east-1",
    "revisionKey": "FAKE_SHA",
    "secretAccessKey": "FAKE_AWS_SECRET",
  },
}
`;

exports[`Given the correct process.env for a production deployTarget generateStorybookDeployConfig matches snapshot using travis env vars 1`] = `
Object {
  "build": Object {
    "environment": "production",
    "outputPath": "./storybook-static/",
  },
  "display-revisions": Object {
    "amount": 10,
    "revisions": [Function],
  },
  "github-deployment-status": Object {
    "environment": "stories",
    "org": "zenefits",
    "ref": "FAKE_TRAVIS_PULL_REQUEST_SHA",
    "repo": "z-frontend",
    "targetUrl": "http://ui.zenefits.com/app/stories/index.html:FAKE_TRAVIS_COMMIT",
    "token": "FAKE_GITHUB_TOKEN",
  },
  "pipeline": Object {
    "alias": Object {
      "s3-index": Object {
        "as": Array [
          "s3-index",
          "s3-iframe",
        ],
      },
    },
  },
  "revision-data": Object {
    "type": "git-commit",
  },
  "s3": Object {
    "accessKeyId": "FAKE_AWS_ACCESS_KEY_ID",
    "bucket": "ui.zenefits.com",
    "filePattern": "{**,docs}/*.{js,css,png,gif,ico,jpg,map,xml,txt,svg,swf,eot,ttf,woff,woff2,otf}",
    "prefix": "app/stories",
    "region": "us-east-1",
    "secretAccessKey": "FAKE_AWS_SECRET",
  },
  "s3-iframe": Object {
    "accessKeyId": "FAKE_AWS_ACCESS_KEY_ID",
    "allowOverwrite": true,
    "bucket": "ui.zenefits.com",
    "filePattern": "iframe.html",
    "prefix": "app/stories",
    "region": "us-east-1",
    "revisionKey": "FAKE_TRAVIS_COMMIT",
    "secretAccessKey": "FAKE_AWS_SECRET",
  },
  "s3-index": Object {
    "accessKeyId": "FAKE_AWS_ACCESS_KEY_ID",
    "allowOverwrite": true,
    "bucket": "ui.zenefits.com",
    "prefix": "app/stories",
    "region": "us-east-1",
    "revisionKey": "FAKE_TRAVIS_COMMIT",
    "secretAccessKey": "FAKE_AWS_SECRET",
  },
}
`;
